app/controllers/categories_controller.rb

```diff
@@ -0,0 +1,36 @@
+class CategoriesController < ApplicationController
+before_action :require_login, only: [:new, :create, :edit, :update]
```

Comment:
It might be easier and shorter to write `before_action :require_login, except: [:index, :show]`